### ISSUE TITLE
Next paper candidates: 2 Nov

### ISSUE THREAD

**Issue by bzz** [src-d/reading-club#11]
# Next paper candidates
Let's propose papers to study next! All papers mentioned in the comments of this issue will be listed in the next vote.

## Last session runner-up(s)
- "Learning to Represent Edits" https://openreview.net/pdf?id=BJl6AjC5F7 
   >> Programming languages are emerging as a challenging and interesting domain for machine learning. A core task, which has received significant attention in recent years, is building generative models of source code. However, to our knowledge, previous generative models have always been framed in terms of generating static snapshots of code. In this work, we instead treat source code as a dynamic object and tackle the problem of modeling the edits that software developers make to source code files. This requires extracting intent from previous edits and leveraging it to generate subsequent edits. We develop several neural networks and use synthetic data to test their ability to learn challenging edit patterns that require strong generalization. We then collect and train our models on a large-scale dataset consisting of millions of fine-grained edits from thousands of Python developers.
   >
   > TL;DR: Neural networks for source code that model changes being made to the code-base rather than static snapshots of code.
   
- "Cuckoo Filter: Better Than Bloom" https://www.cs.cmu.edu/~dga/papers/cuckoo-conext2014.pdf 
   > Cuckoo filters provide the ﬂexibility to add and remove items dynamically. A cuckoo filter is based on cuckoo hashing (and therefore named as cuckoo filter). It is essentially a cuckoo hash table storing each key's fingerprint.

**Comment by m09** [comment]
[Ask the Right Questions: Active Question Reformulation with Reinforcement Learning](https://ai.google/research/pubs/pub46733.pdf)

> We frame Question Answering (QA) as a Reinforcement Learning task, an approach that we call Active Question Answering. We propose an agent that sits between the user and a black box QA system and learns to reformulate questions to elicit the best possible answers. The agent probes the system with, potentially many, natural language reformulations of an initial question and aggregates the returned evidence to yield the best answer. The reformulation system is trained end-to-end to maximize answer quality using policy gradient. We evaluate on SearchQA, a dataset of complex questions extracted from Jeopardy!. The agent outperforms a state-of-the-art base model, playing the role of the environment, and other benchmarks. We also analyze the language that the agent has learned while interacting with the question answering system. We find that successful question reformulations look quite different from natural language paraphrases. The agent is able to discover non-trivial reformulation strategies that resemble classic information retrieval techniques such as term re-weighting (tf-idf) and stemming.

**Comment by bzz** [comment]
The poll is on https://sourced-community.slack.com/archives/CCLPL1WQH/p1540550576002000

**Comment by bzz** [comment]
and the winner is
> Learning to Represent Edits https://openreview.net/pdf?id=BJl6AjC5F7    `4`

#17 adds it to readme.
